File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/MonsterInfo.ts
import { ILoader } from "../../Loader/ILoader";
import { AbstractPower } from "../../Power/AbstractPower";

export abstract class MonsterInfo{
    abstract get Loader() : ILoader;
    abstract get MaxHealth():number;
    abstract get Name():string
    
    get InitPower():Array<AbstractPower>{
        return new Array<AbstractPower>();
    }
    constructor(public X:number,public Y:number) {
    }
    
    toString(){
        return this.Name;
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/MonsterInfo.cs
using System.Collections.Generic;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster
{
    public abstract class MonsterInfo
    {
        public abstract ILoader Loader { get; }
        public  abstract int MaxHealth { get; }
        public abstract string Name { get; }
    
        public readonly List<AbstractPower> InitPower = new List<AbstractPower>();
        public Vector2 Pos;
        public MonsterInfo(Vector2 pos)
        {
            Pos = pos;
        }

        public override string ToString()
        {
            return this.Name;
        }
    }
}